### DISCUSSION TITLE
unable to see stress results

### DISCUSSION THREAD

**Discussion by indominus-hub** [pyansys/pymapdl-reader#144] [Q&A] [answered]
Dear all,

I have tried to use PyANSYS to convert a .rst into a .vtk (readable by Paraview)

Nodal results are well displayed (displacements) but strain and stress are displayed uniformly in a NaN colour (but the max and min values are there in all directions, I can see it in the vtk opening it as a text file, but seems something is wrong with the display).

I did:

rst.save_as_vtk('results.vtk', [0], ['ENF', 'ENS', 'ENS'])

Any hints will be really appreciated

**Comment by akaszynski** [comment]
Hi @indominus-hub,

[pydpf-core](https://github.com/pyansys/pydpf-core) supports exporting results to vtk, but if you need to use `pymapdl-reader`, please upload the result file here and I'll take a look at it. If you're unable to upload it, email me at alexander.kaszynski@ansys.com and I'll see what I can do.

**Reply by indominus-hub** [reply]
Hi,

Unfortunately, I am not allowed to share the files (property of a third party.

I have used pymapdl reader for importing a .rst. The results are displayed properly with for example:

result.plot_principal_nodal_stress(
    0, "SEQV", show_displacement=True, displacement_factor=1
)

But when I try to export to vtk/vtu...it is done and nodal displacements are properly shown, on the contrary many NaN for stress and strain.

Any hints?

**Reply by indominus-hub** [reply]
BTW same behaviour with beam example:

https://reader.docs.pyansys.com/user_guide/loading_results.html#exporting-to-paraview

I tested it in several Paraview (5.9, 5.10) releases and it doesn't work

**Reply by akaszynski** [reply, marked as answer]
I know the issue. You'll want to load in the file, convert all the elements to their linear elements using [linear_copy](https://docs.pyvista.org/api/core/_autosummary/pyvista.UnstructuredGrid.linear_copy.html#linear-copy), and then output it again via:

```py
import pyvista as pv
mesh = pv.read(<file you output>)
mesh.linear_copy().save('output.vtu')
```

The issue is that stresses are not evaluated at the midside nodes but those values are included in the output file. Best to simply ignore then using `linear_copy`.

**Comment by indominus-hub** [comment]
Seems that it works!

Thanks a lot for the support, this was very helpful!
